File: src/main/java/com/cricmanage/model/GamePerformance.java
package com.cricmanage.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "game_performances")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GamePerformance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String username;
    
    @Column(name = "season_beaten")
    private Integer seasonBeaten;
    
    private Integer wins;
    private Integer losses;
    
    @Column(name = "final_position")
    private Integer finalPosition;
    
    private Boolean champion;
    private String mvp;
    
    @Column(name = "biggest_bargain")
    private String biggestBargain;
    
    @Column(name = "worst_pick")
    private String worstPick;
    
    private LocalDateTime timestamp = LocalDateTime.now();
}

File: lib/models/game_performance.dart
class GamePerformance {
  final int? id;
  final String? username;
  final int seasonBeaten;
  final int wins;
  final int losses;
  final int finalPosition;
  final bool champion;
  final String mvp;
  final String biggestBargain;
  final String worstPick;
  final String? timestamp;

  GamePerformance({
    this.id,
    this.username,
    required this.seasonBeaten,
    required this.wins,
    required this.losses,
    required this.finalPosition,
    required this.champion,
    required this.mvp,
    required this.biggestBargain,
    required this.worstPick,
    this.timestamp,
  });

  factory GamePerformance.fromJson(Map<String, dynamic> json) {
    return GamePerformance(
      id: json['id'] as int?,
      username: json['username'] as String?,
      seasonBeaten: json['seasonBeaten'] as int? ?? json['season_beaten'] as int? ?? 0,
      wins: json['wins'] as int? ?? 0,
      losses: json['losses'] as int? ?? 0,
      finalPosition: json['finalPosition'] as int? ?? json['final_position'] as int? ?? 0,
      champion: json['champion'] as bool? ?? false,
      mvp: json['mvp'] as String? ?? 'None',
      biggestBargain: json['biggestBargain'] as String? ?? json['biggest_bargain'] as String? ?? 'None',
      worstPick: json['worstPick'] as String? ?? json['worst_pick'] as String? ?? 'None',
      timestamp: json['timestamp'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'seasonBeaten': seasonBeaten,
      'wins': wins,
      'losses': losses,
      'finalPosition': finalPosition,
      'champion': champion,
      'mvp': mvp,
      'biggestBargain': biggestBargain,
      'worstPick': worstPick,
    };
  }
}
